GoogleForCreators/web-stories-wp
# Dashboard: Keyboard Navigation for Template Detail View

## Summary
Enables users to use the arrow keys to tab through the detail template pages grid. 

![details with keyboard outline](https://user-images.githubusercontent.com/10720454/91499794-f9071f80-e876-11ea-811f-d07ec2027188.gif)



## Relevant Technical Choices

- Took heavily from `edit-story/components/canvas/gridview/useGridViewKeys` to create a custom hook for grid navigation that is not tied to context so that we can reuse it on the different grid cases we have in the dashboard. All I did was update names to be a little more generic since I'll be resusing this for grids containing full stories and just logos rather than only story pages. 
- Also stole from the existing Karma tests to navigate grid with keyboard that exists in `edit-story/components/canvas/karma/gridView.karma.js`

Otherwise, the updates are associated with configuring the card gallery with what the editor did with their grid preview - so buttons as containers and controlling tab index. 

## To-do


## User-facing changes

Now when you navigate to a template detail page you should be able to use the tab key to reach the preview grid and then once there, use the arrow keys to go right/left/down/up. Hit `Enter` to select a new preview page. Hit `Tab` to exit the grid and go on your merry way. 

## Testing Instructions

Go to template detail page in dashboard, tab 4 times. You should see a slight change in the 'active' border of the smaller grid item that's selected. Now you can use your arrow keys to navigate the grid. When you want to change the page you are actively previewing hit 'enter'. 

---



Addresses #4202
